scripts/operations/gateway-test/uan-gateway-test.sh

```diff
@@ -0,0 +1,104 @@
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+if [ $# -ne 1 ]; then
+  echo "usage: $0 <uan-hostname>"
+  exit 1
+fi
+
+UAN_NAME=$1
+
+# Make sure we can resolve the UAN Name
+if [ $(dig +short ${UAN_NAME} | wc -l) -lt 1 ]; then
+    echo "ERROR: Unknown UAN ${UAN_NAME}"
+    exit 1
+fi
+
+# Make sure the cray, craysys, kubectl, and jq commands are available
+
+craysys type get > /dev/null 2>&1
+if [ $? -ne 0 ]; then
+  echo "ERROR: craysys command is not available"
+  exit 1
+fi
+
+kubectl version > /dev/null 2>&1
+if [ $? -ne 0 ]; then
+  echo "ERROR: kubectl command is not available"
+  exit 1
+fi
+
+which jq > /dev/null 2>&1
+if [ $? -ne 0 ]; then
+  echo "ERROR: jq command is not available"
+  exit 1
+fi
+
+# Get the Base Directory
+BASEDIR=$(dirname $0)
+
+# Get the SYSTEM_DOMAIN from cloud-init 
+SYSTEM_NAME=$(craysys metadata get system-name)
+SITE_DOMAIN=$(craysys metadata get site-domain)
+SYSTEM_DOMAIN=${SYSTEM_NAME}.${SITE_DOMAIN}
+echo "System domain is ${SYSTEM_DOMAIN}"
+
+# Get a token to talk to SLS
+export TOKEN=$(curl -s -k -S -d grant_type=client_credentials -d client_id=admin-client -d client_secret=`kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d` https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
+
+# Get the USER_NETWORK from SLS
+USER_NETWORK=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/dumpstate | jq -r '.Networks.BICAN.ExtraProperties.SystemDefaultRoute' | tr '[:upper:]' '[:lower:]')
+echo "User Network on ${SYSTEM_NAME} is ${USER_NETWORK}"
+
+# Get the ADMIN_SECRET
+ADMIN_SECRET=$(kubectl get secrets admin-client-auth -ojsonpath='{.data.client-secret}' | base64 -d)
+if [[ -z ${ADMIN_SECRET} ]]; then
+    echo "ERROR: Failed to retrieve admin client secret"
+    exit 1
+fi
+echo "Got admin client secret"
+
+# Prepare the run script
+cat > /tmp/run-gateway-test-${UAN_NAME}.sh <<EOF
+#!/bin/bash
+    
+export ADMIN_CLIENT_SECRET=$ADMIN_SECRET
+
+echo "Running gateway tests on the UAN...(this may take 1-2 minutes)"
+echo "/root/gateway-test.py ${SYSTEM_DOMAIN} uan ${USER_NETWORK}"
+/root/gateway-test.py ${SYSTEM_DOMAIN} uan ${USER_NETWORK}
+EOF
+
+chmod a+x /tmp/run-gateway-test-${UAN_NAME}.sh
+
```

Comment:
As with the other test, I'd suggest possibly checking the return code of the scp command, and modifying the ssh command so that we can tell if the ssh itself failed (versus one of the remote commands failing. But not a big deal, of course.

Reply:
fixed